File: thesum.c
/*leggere e aggiornare un file di testo con nuovi valori*/
#include <stdio.h>

int main () 
{
    int valore = 0,somma = 0;
    FILE *file;
    
    printf("inserisci un valore intero");
    scanf("%d",&valore);
   
    file = fopen("file", "r");
    if (!file) 
        printf("file inesistente: creazione in corso");

    else 
        fscanf(file, "%d", &somma);
    
    fclose(file);

    somma = somma + valore;

    file = fopen("file", "w");
    fprintf(file, "%d", somma);

    fclose(file);

    return 0;

}
    

    

File: thesum.py
import os 

file = 'file'



if not os.path.isfile(file):
    print("file inesistente o altro: creazione in corso")
    somma = 0
else:
    f = open (file, 'r')
    valore_letto = f.readline()
    somma = int(valore_letto)
    f.close()

valore = int(input ("inserisci un valore"))

somma += valore

f = open (file, 'w')
f.write(str(somma))
f.close()